r5/utils.bal

```diff
@@ -0,0 +1,237 @@
+public isolated function clientErrorToFhirError(http:ClientError clientError) returns FHIRError {
+    if (clientError is http:ClientRequestError) {
+        // handle 4xx errors
+        string errorMsg = getErrorMessage(clientError);
+        return createFHIRError(errorMsg, CODE_SEVERITY_ERROR, TRANSIENT_EXCEPTION,
+            httpStatusCode = clientError.detail().statusCode);
+    }
+    else {
+        // handle other errors
+        return createFHIRError(clientError.message(), CODE_SEVERITY_ERROR,
+            TRANSIENT_EXCEPTION, httpStatusCode = http:STATUS_INTERNAL_SERVER_ERROR);
+    }
+}
+
+# Create a FHIR Bundle.
+#
+# + t - a BundleType.
+# + resources - array of FHIR resources.
+# + return - Return a FHIR Bundle.
+public isolated function createFhirBundle(BundleType t, DomainResource[] resources) returns Bundle {
+    Bundle bundle = {
+        'type: t
+    };
+    BundleEntry[] entry = [];
+    foreach var res in resources {
+        entry.push({
+            'resource: res
+        });
+    }
+    bundle.entry = entry;
+    return bundle;
+}
+
+public isolated function urlEncodeFhirSearchParameters(map<RequestSearchParameter[] & readonly> & readonly params)
+        returns string|FHIRError {
+    string|error encoded = url:encode(params.toJsonString(), "UTF-8");
+    if (encoded is error) {
+        string errorMsg = "Error occurred while encoding the search parameters.";
+        log:printError(errorMsg, 'error = encoded);
+        return createFHIRError(errorMsg, CODE_SEVERITY_FATAL, INVALID, httpStatusCode = http:STATUS_BAD_REQUEST);
+    }
+    return encoded;
+}
+
+public isolated function decodeFhirSearchParameters(string urlEncodedParams)
+        returns map<RequestSearchParameter[] & readonly>|FHIRError {
+    string|error paramsDecoded = url:decode(urlEncodedParams, "UTF-8");
+    if (paramsDecoded is error) {
+        string errorMsg = "Error occurred while decoding the search parameters.";
+        log:printError(errorMsg, 'error = paramsDecoded);
+        return createFHIRError(errorMsg, CODE_SEVERITY_FATAL, INVALID, httpStatusCode = http:STATUS_BAD_REQUEST);
+    }
+    map<RequestSearchParameter[] & readonly>|error arr = paramsDecoded.fromJsonStringWithType();
+    if (arr is error) {
+        string errorMsg = "Error occurred while decoding the search parameters to an array.";
+        log:printError(errorMsg, 'error = arr);
+        return createFHIRError(errorMsg, CODE_SEVERITY_FATAL, INVALID, httpStatusCode = http:STATUS_BAD_REQUEST);
+    }
+    return arr;
+}
+
+# Maps a given FHIR primitive type to a corresponding FHIR search parameter type.
+#
+# + primitiveType - The FHIR primitive type to be mapped
+# + return - The corresponding FHIR search parameter type or a `FHIRError` if the primitive type does not 
+# correspond to a search parameter type
+public isolated function primitiveTypeToSearchParamType(string primitiveType) returns FHIRSearchParameterType|FHIRError {
+    match primitiveType {
```

Comment:
Missing types (but these are marked as `Not used in search`);
1. [markdown](https://hl7.org/fhir/R5/datatypes.html#markdown)
2. [time](https://hl7.org/fhir/R5/datatypes.html#time)

Reply:
Seems these types are not reflected in search param types? https://hl7.org/fhir/R5/searchparameter-registry.html